opentreemap/treemap/tests/test_udfs.py

```diff
@@ -1306,6 +1306,46 @@ def test_delete_cudf_deletes_mobile_api_field_group(self):
         self.assertEquals(1, len(updated_instance.mobile_api_fields))
 
 
+class ProtectedUDFTestCase(OTMTestCase):
+    def setUp(self):
+        self.p = Point(0, 0)
+        self.instance = make_instance(point=self.p)
+        self.commander_user = make_commander_user(self.instance)
+        set_write_permissions(self.instance, self.commander_user,
+                              'Plot', ['udf:Test choice'])
+
+        self.udf_def = UserDefinedFieldDefinition.objects.create(
+            instance=self.instance,
+            model_type='Plot',
+            datatype=json.dumps({'type': 'choice',
+                                 'protected_choices': ['a'],
+                                 'choices': ['b', 'c']}),
+            iscollection=False,
+            is_protected=True,
+            name='Test choice')
+
+        self.plot = Plot(geom=self.p, instance=self.instance)
+        self.plot.save_with_user(self.commander_user)
+
+    def test_cannot_delete_protected_udf(self):
+        self.assertRaises(AuthorizeException, lambda: self.udf_def.delete())
+
+    def test_can_pick_protected_udf_choice(self):
+        # Assert that ValidationError is not raised
+        self.plot.udfs['Test choice'] = 'a'
+        self.plot.save_with_user(self.commander_user)
+
+    def test_cannot_modify_protected_udf_choice(self):
+        def delete_choice():
+            self.udf_def.delete_choice('a')
+
+        def update_choice():
+            self.udf_def.update_choice('a', 'z')
+
+        self.assertRaises(ValidationError, delete_choice)
+        self.assertRaises(ValidationError, update_choice)
```

Comment:
I meant more another case to test, I don't care about the formatting.

FWIW, I usually use the decorator version of `assertRaises`:
```python
with self.assertRaises(AnException) as e:
    call_the_function(bad_data)

self.assertEquals(e.message, 'You gave bad data')